packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -916,7 +916,20 @@ Future<void> _writeMacOSPluginRegistrant(FlutterProject project, List<Plugin> pl
 }
 
 Future<void> _writeWindowsPluginFiles(FlutterProject project, List<Plugin> plugins) async {
-  final List<Map<String, dynamic>> windowsPlugins = _extractPlatformMaps(plugins, WindowsPlugin.kConfigKey);
+  // Plugins without a 'pluginClass' definition shouldn't create native cpp files. The plugin might
+  // be a pure Dart implementation, and the class will be defined by 'dartPluginClass'.
+  final List<Plugin> nativeWindowsPlugins = plugins.where((final Plugin element) {
```

Comment:
no need to use `final` for parameters ... its just too verbose if we did that everywhere
```suggestion
  final List<Plugin> nativeWindowsPlugins = plugins.where((Plugin element) {
```

Reply:
Done